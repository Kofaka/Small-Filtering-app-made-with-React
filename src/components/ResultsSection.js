import React from 'react';

import './ResultsSection.scss';

const ResultsSection = ({items = []}) => {
    const ResultItem = ({item}) => (
        <li className="results-section__item">
            <span className="results-section__item__title">{item.title}</span>
            <span className="results-section__item__description">{item.tags.join(', ')}</span>
        </li>
    );
    return (
        <ul className="results-section">
            {items.map(item => <ResultItem item={item} key={item.id}/>)}
        </ul>
    );
};

export default ResultsSection;
