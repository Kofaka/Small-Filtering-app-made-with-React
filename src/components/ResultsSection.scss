.results-section {
  display: grid;
  grid-template-columns: 50% 50%;
  grid-template-rows: 100%;
  grid-gap: 1.5em;
  margin: 0;
  padding: 0 1.5em;

  &__item {
    margin: 0 1.5em 1em 0;
    padding: 0;

    &__title {
      display: block;
      text-transform: uppercase;
      font-size: larger;
      font-weight: bold;
    }

    &__description {
      display: block;
      text-transform: capitalize;
      font-size: x-small;
    }
  }
}
