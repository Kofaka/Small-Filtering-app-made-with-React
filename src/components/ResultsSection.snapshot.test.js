import React from 'react';
import expect from 'expect';
import renderer from 'react-test-renderer';

import ResultsSection from './ResultsSection';

const setup = (propsOverrides) => {
    const props = Object.assign({
        items: [
            {
                title: 'SDK 1',
                id: 'sdk1',
                tags: ['tag-name-1', 'tag-name-2'],
            },
            {
                title: 'SDK 2',
                id: 'sdk2',
                tags: ['tag-name-3', 'tag-name-5'],
            },
            {
                title: 'SDK 3',
                id: 'sdk3',
                tags: ['tag-name-2', 'tag-name-3', 'tag-name-4'],
            }
        ],
    }, propsOverrides);

    return {
        props
    };
};


describe('Component: ResultsSection', () => {
    it('renders without crashing', () => {
        const {props} = setup();
        const resultsSection = renderer
            .create(<ResultsSection {...props} />)
            .toJSON();
        expect(resultsSection).toMatchSnapshot();
    });
});

