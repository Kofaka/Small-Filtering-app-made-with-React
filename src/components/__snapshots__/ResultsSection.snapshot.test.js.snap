// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ResultsSection renders without crashing 1`] = `
<ul
  className="results-section"
>
  <li
    className="results-section__item"
  >
    <span
      className="results-section__item__title"
    >
      SDK 1
    </span>
    <span
      className="results-section__item__description"
    >
      tag-name-1, tag-name-2
    </span>
  </li>
  <li
    className="results-section__item"
  >
    <span
      className="results-section__item__title"
    >
      SDK 2
    </span>
    <span
      className="results-section__item__description"
    >
      tag-name-3, tag-name-5
    </span>
  </li>
  <li
    className="results-section__item"
  >
    <span
      className="results-section__item__title"
    >
      SDK 3
    </span>
    <span
      className="results-section__item__description"
    >
      tag-name-2, tag-name-3, tag-name-4
    </span>
  </li>
</ul>
`;
